logstash-core/lib/logstash/runner.rb

```diff
@@ -148,25 +148,32 @@ def run(args)
     begin
       LogStash::SETTINGS.from_yaml(LogStash::SETTINGS.get("path.settings"))
     rescue => e
-      @logger.subscribe(STDOUT)
-      @logger.warn("Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options")
-      @logger.fatal("Failed to load settings file from \"path.settings\". Aborting...", "path.settings" => LogStash::SETTINGS.get("path.settings"), "exception" => e.class, "message" => e.message)
+      $stderr.puts "ERROR: Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options"
+      $stderr.puts "ERROR: Failed to load settings file from \"path.settings\". Aborting... path.setting=#{LogStash::SETTINGS.get("path.settings")}, exception=#{e.class}, message=>#{e.message}"
       exit(-1)
     end
 
+    # Configure Logstash logging facility, this need to be done before everything else to
+    # make sure the logger has the correct settings and the log level is correctly defined.
+    # TODO(talevy): cleanly support `path.logs` setting in log4j
+    log4j_config_location = setting("path.settings") + "/log4j2.properties"
+    unless java.lang.System.getProperty("log4j.configurationFile")
+      LogStash::Logging::Logger::initialize(log4j_config_location)
+    end
```

Comment:
To inject `path.log` and `log.level` as log4j2 `Lookup`, I think we could do it easily like this :
- create a java class like this : [LogstashLookup.java](https://gist.github.com/fbaligand/bedeb87a18b876f1d057d43ee171e05c)
- just before log4j initialization, call `LogstashLookup.setLogstashSettings(Logstash::SETTINGS.to_hash)`
- in `log4j2.properties`, add references `${logstash:path.log}` and `${logstash:log.level}`

Great, no ? :)